File: src/drizzle/schema/student.ts
import { integer, varchar, pgTable, bigint } from "drizzle-orm/pg-core";
import { teacher } from "./teacher"

export type Student = typeof student.$inferSelect;

export const student = pgTable('student', {
    id: bigint({ mode: "number" }).primaryKey().generatedAlwaysAsIdentity(),
    firstName: varchar("first_name").notNull(),
    lastName: varchar("last_name"),
    grade: integer(),
    teacherId: bigint("teacher_id", {mode: "number"}).references(() => teacher.id)
})

File: StudentRun/StudentRun.Server/Models/Student.cs
namespace StudentRun.Server.Models;

public partial class Student
{
    public long Id { get; set; }

    public string FirstName { get; set; } = null!;

    public string? LastName { get; set; }

    public int? Grade { get; set; }

    public int Laps { get; set; } = 0;

    public double Miles { get; set; } = 0;

    public DateTime JoinedDate { get; set; }

    public long? TeacherId { get; set; }

    public virtual Teacher? Teacher { get; set; }
}
